stdlib/public/core/FlatMap.swift

```diff
@@ -24,6 +24,25 @@ extension LazySequenceProtocol {
     FlattenSequence<LazyMapSequence<Elements, SegmentOfResult>>> {
     return self.map(transform).flatten()
   }
+  
+  /// Returns a `LazyMapSequence` containing the concatenated non-nil
+  /// results of mapping transform over this `Sequence`.
+  ///
+  /// Use this method to receive only nonoptional values when your
+  /// transformation produces an optional value.
+  ///
+  /// - Parameter transform: A closure that accepts an element of this
+  /// sequence as its argument and returns an optional value.
+  @warn_unused_result
+  public func flatMap<U>(
```

Comment:
Could you change `U` to `ElementOfResult`?

If you feel like doing so, could you also change the names of generic type parameters in the eager `flatMap` overloads (`stdlib/public/core/SequenceAlgorithms.swift.gyb`) to `SegmentOfResult` and `ElementOfResult`?